### PR TITLE
Improve #596

### PR DESCRIPTION
**Pull request by syohex**
## Description

Improve #596

- Refactoring code
- Better naming. (foldable block feature is not only gfm feature)
-  Add some flows
- Add unit test

## Related Issue

#595, #596 CC: @psibi 

## Type of Change

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING.md** document.
- [x] I have updated the documentation in the **README.md** file if necessary.
- [x] I have added an entry to **CHANGES.md**.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed (using `make test`).

### CODE REVIEW

**Comment by psibi** [comment]
The changes LGTM. Thank  you!

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
         `markdown-disable-tooltip-prompt`.
     -   Introduce `markdown-ordered-list-enumeration` variable [GH-587][]
     -   Search wiki link under project
+    -   Add `markdown-insert-gfm-foldable-block` function [GH-596][]
```
Comment (psibi) [inline CHANGES.md:15]: Missed while reviewing the first time: I guess it shouldn't have `gfm` since you renamed it ?

Comment (syohex) [reply CHANGES.md:15]: sorry I'll fix it later

Comment (syohex) [reply CHANGES.md:15]: https://github.com/jrblevin/markdown-mode/commit/d8267c17eb14d3db2ebd079094df524b547607b1

Comment (psibi) [reply CHANGES.md:15]: No worries. :-) Thanks for adding the test case and making the code better!

**Pull request merged**